src/core/environment.go

```diff
@@ -561,19 +561,64 @@ func diskPartitions() (partitions []*proto.DiskPartition) {
 }
 
 func releaseInfo() (release *proto.ReleaseInfo) {
-	hostInfo, err := host.Info()
+	const osReleaseFile =  "/etc/os-release"
+	osRelease, err := getOsRelease(osReleaseFile)
 	if err != nil {
-		log.Errorf("Could not read release information for host: %v", err)
-		return &proto.ReleaseInfo{}
+		hostInfo, err := host.Info()
+		if err != nil {
+			log.Errorf("Could not read release information for host: %v", err)
+			return &proto.ReleaseInfo{}
+		}
+
+		return &proto.ReleaseInfo{
+			VersionId: hostInfo.PlatformVersion,
+			Version:   hostInfo.KernelVersion,
+			Codename:  hostInfo.OS,
+			Name:      hostInfo.PlatformFamily,
+			Id:        hostInfo.Platform,
+		}
 	}
+	return osRelease
+}
 
-	return &proto.ReleaseInfo{
-		VersionId: hostInfo.PlatformVersion,
-		Version:   hostInfo.KernelVersion,
-		Codename:  hostInfo.OS,
-		Name:      hostInfo.PlatformFamily,
-		Id:        hostInfo.Platform,
+// getOsRelease reads osReleaseFile and returns release information for host.
+// If os.Stat(osReleaseFilePath) does not find file, or
+// ioutil.ReadFile(osReleaseFilePath) fails to read file, an error occurs.
+func getOsRelease(osReleaseFile string) (release *proto.ReleaseInfo, err error) {
+	_ , osReleaseError := os.Stat(osReleaseFile)
+	if os.IsNotExist(osReleaseError) {
+		log.Errorf("Could not find path for os-release file on the host")
+		return &proto.ReleaseInfo{}, errors.New("unable to find " + osReleaseFile)
+	}
+
+	osReleaseInfoMap := map[string]string{}
+
+	data, err := ioutil.ReadFile(osReleaseFile)
+	if err != nil {
+		log.Errorf("Could not read os-release file on the host")
+		return &proto.ReleaseInfo{}, err
 	}
+	scanner := bufio.NewScanner(strings.NewReader(string(data)))
+	for scanner.Scan() {
```

Comment:
A couple things here; the `ioutil` package is deprecated, the useful functions are now in `os`, see note at the top of [`ioutil` docs](https://pkg.go.dev/io/ioutil). We don't need to read the whole file into memory, there are a few other tools we can apply from the standard lib to combine the `ReadFile` and Stat` in one `os.Open` call:

```suggestion
	f, err := os.Open(osReleaseFile)
	if err != nil {
		return nil, fmt.Errorf("release file %s unreadable: %w", osReleaseFile, err)	  
	}
	osReleaseInfoMap := map[string]string{}
	scanner := bufio.NewScanner(f)
	for scanner.Scan() {
```

Reply:
There were usage of `ioutils` in this file, so I followed the convention. 
I will have `os` as suggested.